### PR TITLE
Fixes #232: Update documentation

### LINKED ISSUE #232: Reread and complete documentation
1. Installation incomplete.

2. retest examples

3. Review the sections to assure that order of discussing subjects is correct.

4. Add development documentation similar to pywbem

### PR DESCRIPTION
**Pull request by KSchopmeyer**
At this point I have three references that will not resolve 

1. Introductionline 89

2 and 3 mock_support 

:meth:`pywbem_mock.FakedWBEMConnection.add_cim_objects`

 (:meth:`pywbem_mock.method_callback_interface`)

Note that we included defintions for the sphinx intersphinx extension for pywbem and pywbem_mock and that they work with :class:

Update documentation

1, Add development section. Parallels pywbem

2. Update installation section

3. Minor changes for spelling, grammar

### CODE REVIEW

File: docs/appendix.rst [hunk 1]
```diff
@@ -177,16 +204,16 @@ a single invocation of an easy-to use method
 Profile implementations in a WBEM server are not entirely free when making a
 choice of which methodology to implement:
 
-* Autonomous profiles in a WBEM server must implement the central class
-  methodology, and may in addition implement the new GetCentralInstances
+* Autonomous profiles in a WBEM server must implement the central-class
+  methodology, and may in addition implement the get-central-instances
```
Comment (andy-maier) [inline docs/appendix.rst]: In DSP1033 1.1, it is actually called "GetCentralInstances methodology".

File: docs/appendix.rst [hunk 2]
```diff
@@ -177,16 +204,16 @@ a single invocation of an easy-to use method
...
   methodology.
 
   Note that the scoping class methodology falls together with the
   central class methodology for autonomous profiles, because their scoping
   class is also their central class.
 
 * Component profiles in a WBEM server may implement the central class
-  methodology and the new GetCentralInstances methodology, and must support the
+  methodology and the get-central-instances methodology, and must support the
```
Comment (andy-maier) [inline docs/appendix.rst]: See previous comment on methodology name.

File: docs/development.rst [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+
+        $ make
+
+   displays a list of valid Make targets and a short description of what each
+   target does.
+
+.. _virtual Python environment: https://docs.python-guide.org/en/latest/dev/virtualenvs/
+
+
+.. _`Manual setup on Windows`:
```
Comment (andy-maier) [inline docs/development.rst]: This section basically has nothing to manually set up. If there is nothing, the section can as well be removed.

File: docs/development.rst [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+environments on Windows (such as CygWin, MinGW, Babun, or Gow) or native
+windows. The pywbemtools tests that run on the Appveyor CI use both CygWin
+and native windows.
+
+Note that Unix-like environments on Windows bring their own Python, so double
+check that the active Python environment is the one you want to install to.
+
+
+.. _`Prerequisite operating system packages for development`:
+
```
Comment (andy-maier) [inline docs/development.rst]: This section lists no OS level packages, which is correct. That means the table can go away and most of the text.

However, we may want to refer to pywbem's corresponding section.

File: docs/development.rst [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
++--------------------------+--------------------+----------------------+-------+
+| Linux SUSE family        |                    |                      |       |
+| (SLES, openSUSE)         |                    |                      |       |
++--------------------------+--------------------+----------------------+-------+
+| OS-X                     |                    |                      |       |
++--------------------------+--------------------+----------------------+-------+
+| Windows                  |                    |                      |       |
++--------------------------+--------------------+----------------------+-------+
+
+
```
Comment (andy-maier) [inline docs/development.rst]: From here on down to the end of this file, the package names `pywbem` and `pywbem_mock` are used, when tehy should have been changed to `pywbemtools`. Also in URLs.

File: docs/development.rst [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+The `tests` directory has the following subdirectory structure:
+
+::
+
+    tests
+     +-- unit                Unit tests
+     |    +-- utils               Utility functions used by unit tests
+     +-- manual              Manual tests
+     +-- schema              The CIM schema MOF files used by some MOF tests
```
Comment (andy-maier) [inline docs/development.rst]: The `manual` directory is still named `live_unit` in the repo. Change it in the repo.

File: docs/development.rst [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+3. Manual tests
+
+   There are several Python scripts and shell scripts that can be run manually.
+   The results need to be validated manually.
+
+   These scripts are in the directory:
+
+   ::
+
+       tests/manualtest/
```
Comment (andy-maier) [inline docs/development.rst]: Inconsistent name of the manual test directory: `manual` vs `manualtest`. Decide for one and use consistently.

File: docs/development.rst [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+   Some of the scripts support a `--help` option that informs about their
+   usage.
+
+   The `run_cim_operations.py` script needs a particular MOF file loaded in the
+   repository of the WBEM server that is used for the test. This can be done
+   using the MOF compiler of pywbem:
+
+   ::
+
+       $ mof_compiler -s <target_url> tests/unittest/pywbem/test.mof
```
Comment (andy-maier) [inline docs/development.rst]: There is no `unittest` directory, it is called  `unit`.

File: docs/appendix.rst (hunk 1 above)
Comment (KSchopmeyer) [reply docs/appendix.rst]: Fixed DONE

File: docs/appendix.rst (hunk 2 above)
Comment (KSchopmeyer) [reply docs/appendix.rst]: Fixed. DONE

File: docs/development.rst (hunk 6 above)
Comment (KSchopmeyer) [reply docs/development.rst]: DONE in pr # 230, committed.

File: docs/development.rst (hunk 7 above)
Comment (KSchopmeyer) [reply docs/development.rst]: DONE

File: docs/development.rst (hunk 8 above)
Comment (KSchopmeyer) [reply docs/development.rst]: We will keep the short form unit.

File: docs/development.rst (hunk 4 above)
Comment (KSchopmeyer) [reply docs/development.rst]: Gone. DONE including ref to pywbem section

File: docs/development.rst (hunk 5 above)
Comment (KSchopmeyer) [reply docs/development.rst]: Done. Gone

File: docs/development.rst (hunk 3 above)
Comment (andy-maier) [reply docs/development.rst]: I suggest to ignore this comment for now, and possibly come back to it in a final docs review.

Comment (KSchopmeyer) [reply docs/development.rst]: Completely rewrote

**Pull request merged**